File: tests/minimal_init.lua
local function load_module(module, source, directory)
  local module_dir = directory or ("/tmp/" .. module)
  local directory_exists = vim.fn.isdirectory(module_dir)

  if directory_exists == 0 then
    vim.fn.system({ "git", "clone", source, module_dir })
  end

  vim.opt.rtp:append(module_dir)

  return module_dir
end

local plenary_dir = os.getenv("PLENARY_DIR") or "/tmp/plenary.nvim"
load_module("plenary.nvim", "https://github.com/nvim-lua/plenary.nvim", plenary_dir)
load_module("telescope.nvim", "https://github.com/nvim-telescope/telescope.nvim")
vim.opt.rtp:append(".")
vim.cmd("runtime plugin/plenary.vim")
vim.cmd("runtime plugin/telescope.lua")
require("plenary.busted")

File: tests/minimal_init.ts
const append = (option: any, value: any) => {
  option.append(option, value);
};

const loadModule = (module: string, source: string, directory?: string) => {
  const moduleDir = directory || `/tmp/${module}`;
  const directoryExists = vim.fn.isdirectory(moduleDir);

  if (directoryExists === 0) {
    vim.fn.system(["git", "clone", source, moduleDir]);
  }

  append(vim.opt.rtp, moduleDir);

  return moduleDir;
};

const plenaryDir = os.getenv("PLENARY_DIR") || "/tmp/plenary.nvim";
loadModule(
  "plenary.nvim",
  "https://github.com/nvim-lua/plenary.nvim",
  plenaryDir,
);

const telescopeDir = loadModule(
  "telescope.nvim",
  "https://github.com/nvim-telescope/telescope.nvim",
);

append(vim.opt.rtp, ".");
vim.cmd("runtime plugin/plenary.vim");
vim.cmd("runtime plugin/telescope.lua");
require("plenary.busted");
